File: 11.Stack/Javascript/2.MasaiResturant.js
function stack_operations(operations, n) {
    let stack = [];
    let result = [];

    for (let operation of operations) {
        if (operation[0] === "1") {
            if (stack.length > 0) {
                result.push(stack.pop().toString());
            } else {
                result.push("No Food");
            }
        } else if (operation[0] === "2") {
            const [, x] = operation.split(" ");
            stack.push(parseInt(x));
        }
    }

    return result.join("\n");
}

const operations = ["1", "2 5", "2 7", "2 9", "1", "1"];
const n = 5;

const res = stack_operations(operations, n);
console.log(res);

// Output:
// No Food
// 9
// 7

File: 11.Stack/Python/2.MasaiResturant.py
def stack_operations(operations,n):

    stack = []
    result = []

    for operation in operations:
        if operation[0] == "1":
            if stack:
                result.append(str(stack.pop()))
            else:
                result.append("No Food")
        elif operation[0] == "2":
            _,x = operation.split()
            stack.append(int(x))

    return "\n".join(result)

operations = ["1","2 5", "2 7", "2 9", "1", "1"]
n = 5

res = stack_operations(operations,n)
print(res)
# Output:-
# No Food
# 9
# 7